core/domain/exp_domain_test.py

```diff
@@ -2333,6 +2345,14 @@ def test_serialize_and_deserialize_returns_unchanged_exploration(self):
             exp_domain.Exploration.deserialize(
                 exploration.serialize()).to_dict())
 
+    def test_subtitled_html_proto_is_correct(self):
```

Comment:
Is this a good way to write test?

Reply:
@kevintab95 Any suggestion on this? 

I have a bunch of new functions in `exp_domain` file, which are manually setting all the exploration data to the exploration proto files. Each function is responsible for setting data to its respective field, like subtitled_html or recorded_voiceovers, etc. 

Now, I need to write test cases for all these functions, so what could be a better way to test this whole functionality? 
One thing I was thinking to have a complete exploration dict in the test file which consists of all the interactions data and this could be the one exploration to test all these functions. 

Or the other way is to have particular data for each function and test it as I did in the above implementation for subtitle HTML.

Reply:
Hi @anandwana001, I think what you're doing in this function is reasonable -- i.e. constructing a proto and then asserting various things on it. Does that answer your question?

Also, don't think about "write test cases for functions". Think about what behaviour you want to test, e.g. for the code in this case it would be "check that subtitled HTML proto is constructed correctly". Find a set of behaviours that result in all functions being covered. If you have trouble testing a particular behaviour then we can help with that, but it's hard to comment when you say "write test cases for all these functions". You can see more information on behaviour-driven testing here: https://github.com/oppia/oppia/wiki/Backend-tests (see https://github.com/oppia/oppia/wiki/Backend-tests#guidelines-for-writing-good-tests particularly).

Reply:
(P.S. Kevin is travelling so feel free to ask me instead for now.)